daml-script/converter/src/main/scala/Converter.scala

```diff
@@ -71,8 +71,13 @@ private[daml] object Converter {
       def intoOr[R, L](pf: A PartialFunction R)(orElse: => L): Either[L, R] =
         pf.lift(self) toRight orElse
 
-      def expect[R](name: String, pf: A PartialFunction R): ErrorOr[R] =
-        self.intoOr(pf)(s"Expected $name but got ${self.getClass.getSimpleName}")
+      def expect[R](name: String, pf: A PartialFunction R): ErrorOr[R] = {
+        if (name == self.getClass.getSimpleName) {
```

Comment:
Manages an edge case where the partial function can be unmatched and we would ow generate the confusing "Expected XXX but got XXX"